packages/peregrine/lib/talons/Header/useStoreSwitcher.js

```diff
@@ -0,0 +1,89 @@
+import { useQuery } from '@apollo/client';
+import { useCallback, useMemo } from 'react';
+import { useHistory } from 'react-router-dom';
+import { Util } from '@magento/peregrine';
+const { BrowserPersistence } = Util;
+import { useDropdown } from '@magento/peregrine/lib/hooks/useDropdown';
+
+/**
+ * The useStoreSwitcher talon complements the StoreSwitcher component.
+ *
+ * @param {*} props.getStoreConfig the store switcher data getStoreConfig
+ *
+ * @returns {Object}    talonProps.availableStores - Details about the available store views.
+ * @returns {Boolean}   talonProps.storeMenuIsOpen - Whether the menu that this trigger toggles is open or not.
+ * @returns {Ref}       talonProps.storeMenuRef - A React ref to the menu that this trigger toggles.
+ * @returns {Ref}       talonProps.storeMenuTriggerRef - A React ref to the trigger element itself.
+ * @returns {Function}  talonProps.handleTriggerClick - A function for handling when the trigger is clicked.
+ * @returns {Function}  talonProps.handleSwitchStore - A function for handling when the menu item is clicked.
+ */
+
+export const useStoreSwitcher = props => {
+    const { getStoreConfig } = props;
+    const history = useHistory();
+    const storage = new BrowserPersistence();
```

Comment:
This should be initialized outside of the talon.

Reply:
Done, thank you